scripts/hail_batch/hgdp1kg_tobwgs_nfe/hgdp_1kg_tob_wgs_nfe.py

```diff
@@ -0,0 +1,59 @@
+"""Perform pca on nfe samples from the HGDP,1KG, and tob-wgs dataset"""
+
+import click
+import pandas as pd
+import hail as hl
+
+HGDP1KG_TOBWGS = (
+    'gs://cpg-tob-wgs-analysis/1kg_hgdp_tobwgs_pca/'
+    'v1/hgdp1kg_tobwgs_joined_all_samples.mt'
+)
+
+GNOMAD_LIFTOVER_LOADINGS = 'gs://cpg-reference/gnomad/gnomad_loadings_90k_liftover.ht'
+
+GNOMAD_HGDP_1KG_MT = (
+    'gs://gcp-public-data--gnomad/release/3.1/mt/genomes/'
+    'gnomad.genomes.v3.1.hgdp_1kg_subset_dense.mt'
+)
+
+
+@click.command()
+@click.option('--output', help='GCS output path', required=True)
+def query(output):  # pylint: disable=too-many-locals
+    """Query script entry point."""
+
+    hl.init(default_reference='GRCh38')
+
+    mt = hl.read_matrix_table(HGDP1KG_TOBWGS)
+    mt = mt.annotate_cols(TOB_WGS=mt.s.contains('TOB'))
+    # Get NFE samples only
+    mt = mt.filter_cols(
+        (mt.hgdp_1kg_metadata.population_inference.pop == 'nfe') | (mt.TOB_WGS)
+    )
+    mt_path = f'{output}/hgdp1kg_nfe_samples.mt'
+    if not hl.hadoop_exists(mt_path):
+        mt.write(mt_path)
+
+    # Perform PCA
+    eigenvalues_path = f'{output}/eigenvalues.csv'
+    scores_path = f'{output}/scores.ht'
+    loadings_path = f'{output}/loadings.ht'
+    eigenvalues, scores, loadings = hl.hwe_normalized_pca(
+        mt.GT, compute_loadings=True, k=20
+    )
+    eigenvalues_df = pd.DataFrame(eigenvalues)
+    eigenvalues_df.to_csv(eigenvalues_path, index=False)
+    scores.write(scores_path, overwrite=True)
+    loadings.write(loadings_path, overwrite=True)
+
+    # Get gnomAD allele frequency of variants that aren't in TOB-WGS
+    loadings_gnomad = hl.read_table(GNOMAD_LIFTOVER_LOADINGS).key_by('locus', 'alleles')
+    gnomad_only_variants = loadings_gnomad.anti_join(mt.rows())
+    hgdp_1kg = hl.read_matrix_table(GNOMAD_HGDP_1KG_MT)
+    hgdp_1kg = hgdp_1kg.semi_join_rows(gnomad_only_variants)
+    hgdp_1kg_path = f'{output}/hgdp_1kg_nfe_variants.mt'
```

Comment:
This will write all the samples -- is this really what we want?

How about starting with the loadings, annotating them with the gnomAD AF (including popmax AF) and anything else that might be interesting) and then adding an annotation to indicate whether that variant was present in any of the TOB-WGS samples?

Reply:
Annotating with the TOB-WGS cohort AF might be even more useful.

Reply:
I did what you suggested, adding in the gnomad AF and gnomad popmax AF, then annotating variants that are present in TOB-WGS samples. 

"Annotating with the TOB-WGS cohort AF might be even more useful." - do you know where I could get the TOB-WGS cohort AF?

Reply:
Could you add the gnomAD `nfe` AF as well?

https://hail.is/docs/0.2/aggregators.html#hail.expr.aggregators.call_stats might be the easiest way to compute the TOB-WGS specific AF.

Reply:
I'm not entirely sure about the `nfe` AF. It might be in the `gnomad_freq.AF` field (which we included) but need to confirm. 

Thanks for the link! I'll look into  calculating the TOB-WGS AF 👌